helm/templates/metrics-service.yaml

```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "app.fullname" . }}-metrics
+  name: {{ .Chart.Name | trimSuffix "-chart" | trunc 44 }}-controller-metrics
```

Comment:
can you explain what the error was and how this fixes it ?

Reply:
Because "applicationautoscaling" is such a long name: https://github.com/aws-controllers-k8s/code-generator/pull/195